pkg/api/validation/events.go

```diff
@@ -25,11 +25,20 @@ import (
 // ValidateEvent makes sure that the event makes sense.
 func ValidateEvent(event *api.Event) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
-	if event.Namespace != event.InvolvedObject.Namespace {
-		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.namespace", event.InvolvedObject.Namespace, "namespace does not match involvedObject"))
-	}
+
 	if !util.IsDNSSubdomain(event.Namespace) {
 		allErrs = append(allErrs, errs.NewFieldInvalid("namespace", event.Namespace, ""))
 	}
+	// TODO: event.InvolvedObject is not versioned. References need to be normalizd to internal when converted from external and vice versa.
+	if event.InvolvedObject.Kind == "Minion" || event.InvolvedObject.Kind == "Node" {
```

Comment:
The latest.RESTMapper is somewhat a raw interface for handling rest mapping.  It looks weired that we translate request object in rest level using restmapper, and pass restmapper to validation level to handle referenced object.  I think we should do what @smarterclayton suggested, handle it before validation.  But instead of filtering just for event, can we generalize it?  For example, extend scopeNamer to return reference object scope, etc.  Not very familiar with our rest code, but feel like this is the correct way to go.